### ISSUE TITLE
Script tags while displaying h5p contents

### ISSUE THREAD

**Issue by hoboscientist** [tunapanda/h5p-standalone#18]
Hey, I am new to all this HTML, CSS, JS. I would like to integrate h5p into my webpage but the issue is `<script>H5PIntegration = {"baseUrl":"http:\/\/localhost","url":"\/wp-content\/uploads\/h5p","postUserStatistics":true,"ajax":{"setFinished":"http:\/\/localhost\/wp-admin\/admin-ajax.php?token=64fae661e4&action=h5p_setFinished","contentUserData":"http:\/\/localhost\/wp-admin\/admin-ajax.php?token=7f8db7740e&action=h5p_contents_user_data&content_id=:contentId&data_type=:dataType&sub_content_id=:subContentId"},"saveFreq":false,"siteUrl":"http:\/\/localhost","l10n":{"H5P":{"fullscreen":"Fullscreen","disableFullscreen":"Disable fullscreen","download":"Download","copyrights":"Rights of use","embed":"Embed","size":"Size","showAdvanced":"Show advanced","hideAdvanced":"Hide advanced","advancedHelp":"Include this script on your website if you want dynamic sizing of the embedded content:","copyrightInformation":"Rights of use","close":"Close","title":"Title","author":"Author","year":"Year","source":"Source","license":"License","thumbnail":"Thumbnail","noCopyrights":"No copyright information available for this content.","downloadDescription":"Download this content as a H5P file.","copyrightsDescription":"View copyright information for this content.","embedDescription":"View the embed code for this content.","h5pDescription":"Visit H5P.org to check out more cool content.","contentChanged":"This content has changed since you last used it.","startingOver":"You'll be starting over.","by":"by","showMore":"Show more","showLess":"Show less","subLevel":"Sublevel","confirmDialogHeader":"Confirm action","confirmDialogBody":"Please confirm that you wish to proceed. This action is not reversible.","cancelLabel":"Cancel","confirmLabel":"Confirm","licenseU":"Undisclosed","licenseCCBY":"Attribution","licenseCCBYSA":"Attribution-ShareAlike","licenseCCBYND":"Attribution-NoDerivs","licenseCCBYNC":"Attribution-NonCommercial","licenseCCBYNCSA":"Attribution-NonCommercial-ShareAlike","licenseCCBYNCND":"Attribution-NonCommercial-NoDerivs","licenseCC40":"4.0 International","licenseCC30":"3.0 Unported","licenseCC25":"2.5 Generic","licenseCC20":"2.0 Generic","licenseCC10":"1.0 Generic","licenseGPL":"General Public License","licenseV3":"Version 3","licenseV2":"Version 2","licenseV1":"Version 1","licensePD":"Public Domain","licenseCC010":"CC0 1.0 Universal (CC0 1.0) Public Domain Dedication","licensePDM":"Public Domain Mark","licenseC":"Copyright"}},"hubIsEnabled":true,"user":{"name":"admin","mail":"hoboscientist13@gmail.com"},"core":{"styles":["\/wp-content\/plugins\/h5p\/h5p-php-library\/styles\/h5p.css?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/styles\/h5p-confirmation-dialog.css?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/styles\/h5p-core-button.css?ver=1.10.1"],"scripts":["\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/jquery.js?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p.js?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-event-dispatcher.js?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-x-api-event.js?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-x-api.js?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-content-type.js?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-confirmation-dialog.js?ver=1.10.1","\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-action-bar.js?ver=1.10.1"]},"loadedJs":[],"loadedCss":[],"contents":{"cid-2":{"library":"H5P.InteractiveVideo 1.17","jsonContent":"{\"interactiveVideo\":{\"video\":{\"startScreenOptions\":{\"title\":\"Interactive Video\",\"hideStartTitle\":false,\"copyright\":\"\"},\"textTracks\":[{\"label\":\"Subtitles\",\"kind\":\"subtitles\",\"srcLang\":\"en\"}],\"files\":[{\"path\":\"https:\\\/\\\/www.youtube.com\\\/watch?v=8Yfo_7UrBTo\",\"mime\":\"video\\\/YouTube\",\"copyright\":{\"license\":\"U\"}}]},\"assets\":{\"interactions\":[{\"x\":46.868475991649,\"y\":38.404452690167,\"width\":10,\"height\":10,\"duration\":{\"from\":1,\"to\":11},\"libraryTitle\":\"True\\\/False Question\",\"action\":{\"library\":\"H5P.TrueFalse 1.2\",\"params\":{\"correct\":\"true\",\"l10n\":{\"trueText\":\"True\",\"falseText\":\"False\",\"score\":\"You got @score of @total points\",\"checkAnswer\":\"Check\",\"showSolutionButton\":\"Show solution\",\"tryAgain\":\"Retry\",\"wrongAnswerMessage\":\"Wrong answer\",\"correctAnswerMessage\":\"Correct answer\",\"scoreBarLabel\":\"You got :num out of :total points\"},\"behaviour\":{\"enableRetry\":true,\"enableSolutionsButton\":true,\"disableImageZooming\":false,\"confirmCheckDialog\":false,\"confirmRetryDialog\":false,\"autoCheck\":false},\"confirmCheck\":{\"header\":\"Finish ?\",\"body\":\"Are you sure you wish to finish ?\",\"cancelLabel\":\"Cancel\",\"confirmLabel\":\"Finish\"},\"confirmRetry\":{\"header\":\"Retry ?\",\"body\":\"Are you sure you wish to retry ?\",\"cancelLabel\":\"Cancel\",\"confirmLabel\":\"Confirm\"},\"question\":\"<p>This is an interaction.<\\\/p>\\n\"},\"subContentId\":\"4edd48f6-cd39-4ffe-9e0f-b883eadad0a1\"},\"pause\":false,\"displayType\":\"button\",\"buttonOnMobile\":false,\"adaptivity\":{\"correct\":{\"allowOptOut\":false,\"message\":\"\"},\"wrong\":{\"allowOptOut\":false,\"message\":\"\"},\"requireCompletion\":false},\"label\":\"<p>TEST<\\\/p>\\n\"}]},\"summary\":{\"task\":{\"library\":\"H5P.Summary 1.8\",\"params\":{\"intro\":\"Choose the correct statement.\",\"summaries\":[{\"subContentId\":\"b5872686-f5a6-4042-ac95-3d0b84dd1dbf\",\"tip\":\"\"}],\"overallFeedback\":[{\"from\":0,\"to\":100}],\"solvedLabel\":\"Progress:\",\"scoreLabel\":\"Wrong answers:\",\"resultLabel\":\"Your result\",\"labelCorrect\":\"Correct.\",\"labelIncorrect\":\"Incorrect! Please try again.\",\"labelCorrectAnswers\":\"Correct answers.\",\"tipButtonLabel\":\"Show tip\",\"scoreBarLabel\":\"You got :num out of :total points\",\"progressText\":\"Progress :num of :total\"},\"subContentId\":\"87403620-750b-4535-a56e-929391594825\"},\"displayAt\":3}},\"override\":{\"autoplay\":false,\"loop\":false,\"showBookmarksmenuOnLoad\":false,\"showRewind10\":false,\"preventSkipping\":false,\"deactivateSound\":false},\"l10n\":{\"interaction\":\"Interaction\",\"play\":\"Play\",\"pause\":\"Pause\",\"mute\":\"Mute\",\"unmute\":\"Unmute\",\"quality\":\"Video Quality\",\"captions\":\"Captions\",\"close\":\"Close\",\"fullscreen\":\"Fullscreen\",\"exitFullscreen\":\"Exit Fullscreen\",\"summary\":\"Summary\",\"bookmarks\":\"Bookmarks\",\"defaultAdaptivitySeekLabel\":\"Continue\",\"continueWithVideo\":\"Continue with video\",\"playbackRate\":\"Playback Rate\",\"rewind10\":\"Rewind 10 Seconds\",\"navDisabled\":\"Navigation is disabled\",\"sndDisabled\":\"Sound is disabled\",\"requiresCompletionWarning\":\"You need to answer all the questions correctly before continuing.\",\"back\":\"Back\",\"hours\":\"Hours\",\"minutes\":\"Minutes\",\"seconds\":\"Seconds\",\"currentTime\":\"Current time:\",\"totalTime\":\"Total time:\",\"navigationHotkeyInstructions\":\"Use key k for starting and stopping video at any time\",\"singleInteractionAnnouncement\":\"Interaction appeared:\",\"multipleInteractionsAnnouncement\":\"Multiple interactions appeared.\",\"videoPausedAnnouncement\":\"Video is paused\",\"content\":\"Content\"}}","fullScreen":"1","exportUrl":"\/wp-content\/uploads\/h5p\/exports\/test-2-2.h5p","embedCode":"<iframe src=\"http:\/\/localhost\/wp-admin\/admin-ajax.php?action=h5p_embed&id=2\" width=\":w\" height=\":h\" frameborder=\"0\" allowfullscreen=\"allowfullscreen\"><\/iframe>","resizeCode":"<script src=\"http:\/\/localhost\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-resizer.js\" charset=\"UTF-8\"><\/script>","url":"http:\/\/localhost\/wp-admin\/admin-ajax.php?action=h5p_embed&id=2","title":"Test 2","displayOptions":{"frame":true,"export":true,"embed":true,"copyright":true,"icon":true},"contentUserData":[{"state":"{}"}],"scripts":["\/wp-content\/uploads\/h5p\/cachedassets\/b32b19c89c5ad430f1f75d9e9187dc120640e638.js"],"styles":["\/wp-content\/uploads\/h5p\/cachedassets\/b32b19c89c5ad430f1f75d9e9187dc120640e638.css"]},"cid-1":{"library":"H5P.InteractiveVideo 1.17","jsonContent":"{\"interactiveVideo\":{\"video\":{\"startScreenOptions\":{\"title\":\"Interactive Video\",\"hideStartTitle\":false,\"copyright\":\"\"},\"textTracks\":[{\"label\":\"Subtitles\",\"kind\":\"subtitles\",\"srcLang\":\"en\"}],\"files\":[{\"path\":\"https:\\\/\\\/www.youtube.com\\\/watch?v=jY1szuZy57w\",\"mime\":\"video\\\/YouTube\",\"copyright\":{\"license\":\"U\"}}]},\"assets\":{\"interactions\":[{\"x\":47.807933194154,\"y\":46.103896103896,\"width\":10,\"height\":10,\"duration\":{\"from\":10,\"to\":13},\"libraryTitle\":\"Multiple Choice\",\"action\":{\"library\":\"H5P.MultiChoice 1.10\",\"params\":{\"answers\":[{\"correct\":true,\"tipsAndFeedback\":{\"tip\":\"\",\"chosenFeedback\":\"\",\"notChosenFeedback\":\"\"},\"text\":\"<div>Not Me<\\\/div>\\n\"},{\"correct\":false,\"tipsAndFeedback\":{\"tip\":\"\",\"chosenFeedback\":\"\",\"notChosenFeedback\":\"\"},\"text\":\"<div>Me<\\\/div>\\n\"}],\"overallFeedback\":[{\"from\":0,\"to\":100}],\"UI\":{\"checkAnswerButton\":\"Check\",\"showSolutionButton\":\"Show solution\",\"tryAgainButton\":\"Retry\",\"tipsLabel\":\"Show tip\",\"scoreBarLabel\":\"You got :num out of :total points\",\"tipAvailable\":\"Tip available\",\"feedbackAvailable\":\"Feedback available\",\"readFeedback\":\"Read feedback\",\"wrongAnswer\":\"Wrong answer\",\"correctAnswer\":\"Correct answer\",\"shouldCheck\":\"Should have been checked\",\"shouldNotCheck\":\"Should not have been checked\",\"noInput\":\"Please answer before viewing the solution\"},\"behaviour\":{\"enableRetry\":true,\"enableSolutionsButton\":true,\"type\":\"auto\",\"singlePoint\":false,\"randomAnswers\":true,\"showSolutionsRequiresInput\":true,\"disableImageZooming\":false,\"confirmCheckDialog\":false,\"confirmRetryDialog\":false,\"autoCheck\":false,\"passPercentage\":100,\"showScorePoints\":true},\"confirmCheck\":{\"header\":\"Finish ?\",\"body\":\"Are you sure you wish to finish ?\",\"cancelLabel\":\"Cancel\",\"confirmLabel\":\"Finish\"},\"confirmRetry\":{\"header\":\"Retry ?\",\"body\":\"Are you sure you wish to retry ?\",\"cancelLabel\":\"Cancel\",\"confirmLabel\":\"Confirm\"},\"question\":\"<p>Who is a dick ?<\\\/p>\\n\"},\"subContentId\":\"0bb5efca-2d2f-4297-bd21-6cfd5c5fc617\"},\"pause\":false,\"displayType\":\"button\",\"buttonOnMobile\":false,\"adaptivity\":{\"correct\":{\"allowOptOut\":false,\"message\":\"\"},\"wrong\":{\"allowOptOut\":false,\"message\":\"\"},\"requireCompletion\":false},\"label\":\"\"}]},\"summary\":{\"task\":{\"library\":\"H5P.Summary 1.8\",\"params\":{\"intro\":\"<p>Choose the correct statement.<\\\/p>\\n\",\"overallFeedback\":[{\"from\":0,\"to\":100}],\"solvedLabel\":\"Progress:\",\"scoreLabel\":\"Wrong answers:\",\"resultLabel\":\"Your result\",\"labelCorrect\":\"Correct.\",\"labelIncorrect\":\"Incorrect! Please try again.\",\"labelCorrectAnswers\":\"Correct answers.\",\"tipButtonLabel\":\"Show tip\",\"scoreBarLabel\":\"You got :num out of :total points\",\"progressText\":\"Progress :num of :total\",\"summaries\":[{\"subContentId\":\"53e709f8-6ae8-4e5f-aee7-730d3569b3c1\",\"tip\":\"\"}]},\"subContentId\":\"4938ffea-e386-46ad-8c7b-b5dd3ffa7247\"},\"displayAt\":3}},\"override\":{\"autoplay\":false,\"loop\":false,\"showBookmarksmenuOnLoad\":false,\"showRewind10\":false,\"preventSkipping\":false,\"deactivateSound\":false},\"l10n\":{\"interaction\":\"Interaction\",\"play\":\"Play\",\"pause\":\"Pause\",\"mute\":\"Mute\",\"unmute\":\"Unmute\",\"quality\":\"Video Quality\",\"captions\":\"Captions\",\"close\":\"Close\",\"fullscreen\":\"Fullscreen\",\"exitFullscreen\":\"Exit Fullscreen\",\"summary\":\"Summary\",\"bookmarks\":\"Bookmarks\",\"defaultAdaptivitySeekLabel\":\"Continue\",\"continueWithVideo\":\"Continue with video\",\"playbackRate\":\"Playback Rate\",\"rewind10\":\"Rewind 10 Seconds\",\"navDisabled\":\"Navigation is disabled\",\"sndDisabled\":\"Sound is disabled\",\"requiresCompletionWarning\":\"You need to answer all the questions correctly before continuing.\",\"back\":\"Back\",\"hours\":\"Hours\",\"minutes\":\"Minutes\",\"seconds\":\"Seconds\",\"currentTime\":\"Current time:\",\"totalTime\":\"Total time:\",\"navigationHotkeyInstructions\":\"Use key k for starting and stopping video at any time\",\"singleInteractionAnnouncement\":\"Interaction appeared:\",\"multipleInteractionsAnnouncement\":\"Multiple interactions appeared.\",\"videoPausedAnnouncement\":\"Video is paused\",\"content\":\"Content\"}}","fullScreen":"1","exportUrl":"\/wp-content\/uploads\/h5p\/exports\/test-ashim-1.h5p","embedCode":"<iframe src=\"http:\/\/localhost\/wp-admin\/admin-ajax.php?action=h5p_embed&id=1\" width=\":w\" height=\":h\" frameborder=\"0\" allowfullscreen=\"allowfullscreen\"><\/iframe>","resizeCode":"<script src=\"http:\/\/localhost\/wp-content\/plugins\/h5p\/h5p-php-library\/js\/h5p-resizer.js\" charset=\"UTF-8\"><\/script>","url":"http:\/\/localhost\/wp-admin\/admin-ajax.php?action=h5p_embed&id=1","title":"Test Ashim","displayOptions":{"frame":true,"export":true,"embed":true,"copyright":true,"icon":true},"contentUserData":[{"state":"{}"}],"scripts":["\/wp-content\/uploads\/h5p\/cachedassets\/b30c2b944deb49c9d72c1f9a339974f6787fbab8.js"],"styles":["\/wp-content\/uploads\/h5p\/cachedassets\/b30c2b944deb49c9d72c1f9a339974f6787fbab8.css"]}}};</script><link rel='stylesheet' id='h5p-core-styles-h5p-css'  href='http://localhost/wp-content/plugins/h5p/h5p-php-library/styles/h5p.css?ver=1.10.1' type='text/css' media='all' />`
the above script is being loaded by the browser and is visible when viewing as source.
I would like to somehow store this in a js on the server and then use a src tag to the call the script. 
Hope you understand my issue.
Regards.

**Comment by Jakeii** [comment]
The code above is generated based on the h5p content. As this is an entirely front-end library, additional files can't be generated.
If you have a server component why not use h5p the traditional way for example the WordPress plugin

**Comment by hoboscientist** [comment]
Hey, I am currently using Wordpress plugin. The solution I am looking is for a way to not send the above code into the client side browser or should I say, a way to obfuscate the above underlying code.

**Comment by Jakeii** [comment]
I've just checked and the wordpress plugin also does this, If this is still an issue for you, you'll need to  try opening an issue in https://github.com/h5p/h5p-php-library
